Docker example broken

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```
    Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 21.5.0: Tue Apr 26 21:08:37 PDT 2022; root:xnu-8020.121.3~4/RELEASE_ARM64_T6000
    Binaries:
      Node: 17.6.0
      npm: 8.5.3
      Yarn: 1.22.17
      pnpm: 7.0.0
    Relevant packages:
      next: 12.2.4-canary.2
      eslint-config-next: N/A
      react: 17.0.2
      react-dom: 17.0.2
```

### Which example does this report relate to?

with-docker-multi-env

### What browser are you using? (if relevant)

_No response_

### How are you deploying your application? (if relevant)

_No response_

### Describe the Bug

PR #39072 still doesnt solve issue with broken docker example

```shell
next.js/examples/with-docker-multi-env  canary ---------------------------------------------------------- 4s  03:38:13 PM
> make start-production
docker compose -f docker/production/docker-compose.yml up -d
[+] Running 0/1
 ⠿ with-docker-multi-env-production Error                                                                                           1.5s
[+] Building 1.2s (8/20)
 => [internal] load build definition from Dockerfile                                                                                0.0s
 => => transferring dockerfile: 1.71kB                                                                                              0.0s
 => [internal] load .dockerignore                                                                                                   0.0s
 => => transferring context: 120B                                                                                                   0.0s
 => [internal] load metadata for docker.io/library/node:16-alpine                                                                   1.1s
 => [deps 1/5] FROM docker.io/library/node:16-alpine@sha256:da32af0cf608622b1550678b2552b7d997def7d0ada00e0eca0166ed2ea42186        0.0s
 => [internal] load build context                                                                                                   0.0s
 => => transferring context: 33.38kB                                                                                                0.0s
 => CACHED [deps 2/5] RUN apk add --no-cache libc6-compat                                                                           0.0s
 => CACHED [deps 3/5] WORKDIR /app                                                                                                  0.0s
 => CACHED [deps 4/5] COPY package.json yarn.lock* package-lock.json* pnpm-lock.yaml* ./                                            0.0s
 => ERROR [deps 5/5] RUN   if [ -f yarn.lock ] then yarn --frozen-lockfile;   elif [ -f package-lock.json ] then npm ci;   elif [   0.2s
------
 > [deps 5/5] RUN   if [ -f yarn.lock ] then yarn --frozen-lockfile;   elif [ -f package-lock.json ] then npm ci;   elif [ -f pnpm-lock.yaml ] then yarn global add pnpm && pnpm i;   else echo "Lockfile not found." && exit 1;   fi:
#0 0.174 /bin/sh: syntax error: unexpected "elif" (expecting "then")
------
failed to solve: executor failed running [/bin/sh -c if [ -f yarn.lock ] then yarn --frozen-lockfile;   elif [ -f package-lock.json ] then npm ci;   elif [ -f pnpm-lock.yaml ] then yarn global add pnpm && pnpm i;   else echo "Lockfile not found." && exit 1;   fi]: exit code: 2
make: *** [start-production] Error 17
```

docker version 
```shell
docker version
Client:
 Cloud integration: v1.0.24
 Version:           20.10.17
 API version:       1.41
 Go version:        go1.17.11
 Git commit:        100c701
 Built:             Mon Jun  6 23:04:45 2022
 OS/Arch:           darwin/arm64
 Context:           default
 Experimental:      true

Server: Docker Desktop 4.10.1 (82475)
 Engine:
  Version:          20.10.17
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.17.11
  Git commit:       a89b842
  Built:            Mon Jun  6 23:01:01 2022
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.6.6
  GitCommit:        10c12954828e7c7c9b6e0ea9b0c02b01407d3ae1
 runc:
  Version:          1.1.2
  GitCommit:        v1.1.2-0-ga916309
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### Expected Behavior

docker build example can build without error

### To Reproduce

`make start-production`

--- a/examples/with-docker-compose/next-app/prod-without-multistage.Dockerfile
+++ b/examples/with-docker-compose/next-app/prod-without-multistage.Dockerfile
@@ -6,9 +6,9 @@ WORKDIR /app
 COPY package.json yarn.lock* package-lock.json* pnpm-lock.yaml* ./
 # Omit --production flag for TypeScript devDependencies
 RUN \
-  if [ -f yarn.lock ] then yarn install --frozen-lockfile; \
-  elif [ -f package-lock.json ] then npm ci; \
-  elif [ -f pnpm-lock.yaml ] then yarn global add pnpm && pnpm i; \
+  if [ -f yarn.lock ]; then yarn install --frozen-lockfile; \
+  elif [ -f package-lock.json ]; then npm ci; \
+  elif [ -f pnpm-lock.yaml ]; then yarn global add pnpm && pnpm i; \
   else echo "Lockfile not found." && exit 1; \
   fi
 

--- a/examples/with-docker-compose/next-app/prod.Dockerfile
+++ b/examples/with-docker-compose/next-app/prod.Dockerfile
@@ -7,9 +7,9 @@ WORKDIR /app
 COPY package.json yarn.lock* package-lock.json* pnpm-lock.yaml* ./
 # Omit --production flag for TypeScript devDependencies
 RUN \
-  if [ -f yarn.lock ] then yarn install --frozen-lockfile; \
-  elif [ -f package-lock.json ] then npm ci; \
-  elif [ -f pnpm-lock.yaml ] then yarn global add pnpm && pnpm i; \
+  if [ -f yarn.lock ]; then yarn install --frozen-lockfile; \
+  elif [ -f package-lock.json ]; then npm ci; \
+  elif [ -f pnpm-lock.yaml ]; then yarn global add pnpm && pnpm i; \
   else echo "Lockfile not found." && exit 1; \
   fi
 

--- a/examples/with-docker-multi-env/docker/development/Dockerfile
+++ b/examples/with-docker-multi-env/docker/development/Dockerfile
@@ -8,9 +8,9 @@ WORKDIR /app
 # Install dependencies based on the preferred package manager
 COPY package.json yarn.lock* package-lock.json* pnpm-lock.yaml* ./
 RUN \
-  if [ -f yarn.lock ] then yarn --frozen-lockfile;\
-  elif [ -f package-lock.json ] then npm ci;\
-  elif [ -f pnpm-lock.yaml ] then yarn global add pnpm && pnpm i;\
+  if [ -f yarn.lock ]; then yarn --frozen-lockfile;\
+  elif [ -f package-lock.json ]; then npm ci;\
+  elif [ -f pnpm-lock.yaml ]; then yarn global add pnpm && pnpm i;\
   else echo "Lockfile not found." && exit 1;\
   fi
 

--- a/examples/with-docker-multi-env/docker/production/Dockerfile
+++ b/examples/with-docker-multi-env/docker/production/Dockerfile
@@ -8,9 +8,9 @@ WORKDIR /app
 # Install dependencies based on the preferred package manager
 COPY package.json yarn.lock* package-lock.json* pnpm-lock.yaml* ./
 RUN \
-  if [ -f yarn.lock ] then yarn --frozen-lockfile; \
-  elif [ -f package-lock.json ] then npm ci; \
-  elif [ -f pnpm-lock.yaml ] then yarn global add pnpm && pnpm i; \
+  if [ -f yarn.lock ]; then yarn --frozen-lockfile; \
+  elif [ -f package-lock.json ]; then npm ci; \
+  elif [ -f pnpm-lock.yaml ]; then yarn global add pnpm && pnpm i; \
   else echo "Lockfile not found." && exit 1; \
   fi
 

--- a/examples/with-docker-multi-env/docker/staging/Dockerfile
+++ b/examples/with-docker-multi-env/docker/staging/Dockerfile
@@ -8,9 +8,9 @@ WORKDIR /app
 # Install dependencies based on the preferred package manager
 COPY package.json yarn.lock* package-lock.json* pnpm-lock.yaml* ./
 RUN \
-  if [ -f yarn.lock ] then yarn --frozen-lockfile; \
-  elif [ -f package-lock.json ] then npm ci; \
-  elif [ -f pnpm-lock.yaml ] then yarn global add pnpm && pnpm i; \
+  if [ -f yarn.lock ]; then yarn --frozen-lockfile; \
+  elif [ -f package-lock.json ]; then npm ci; \
+  elif [ -f pnpm-lock.yaml ]; then yarn global add pnpm && pnpm i; \
   else echo "Lockfile not found." && exit 1; \
   fi
 

